store/provide.go

```diff
@@ -104,6 +108,12 @@ func newTransportConfig(in transportConfigIn) (*transportConfig, error) {
 		ItemMaxTTL:              userInputValidation.ItemMaxTTL,
 	}
 
+	var itemDataMaxDepth = defaultItemDataMaxDepth
+	if userInputValidation.ItemDataMaxDepth != nil {
```

Comment:
what if the provided max depth is 0? Should we allow that?

Reply:
I think this ties into my other comment; if any non-nil map at least has a depth of 1, I think we can change `ItemDataMaxDepth` to a `uint` and check that it's not `0`.